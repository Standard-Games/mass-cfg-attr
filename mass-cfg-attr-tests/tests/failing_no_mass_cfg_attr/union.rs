#![deny(unused_must_use, deprecated)]
#![allow(path_statements)] // disable warning for test

#[must_use]
union Union {
    #[deprecated]
    f: f64,

    u: u64,
}

fn main() {
    // Violates unused_must_use and deprecated without mass_cfg_attr
    Union { f: 0.0 };
}
