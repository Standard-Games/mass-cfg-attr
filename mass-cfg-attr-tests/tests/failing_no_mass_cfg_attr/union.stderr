error: use of deprecated field `Union::f`
  --> tests/failing_no_mass_cfg_attr/union.rs:14:13
   |
14 |     Union { f: 0.0 };
   |             ^^^^^^
   |
note: the lint level is defined here
  --> tests/failing_no_mass_cfg_attr/union.rs:1:26
   |
1  | #![deny(unused_must_use, deprecated)]
   |                          ^^^^^^^^^^

error: unused `Union` that must be used
  --> tests/failing_no_mass_cfg_attr/union.rs:14:5
   |
14 |     Union { f: 0.0 };
   |     ^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/failing_no_mass_cfg_attr/union.rs:1:9
   |
1  | #![deny(unused_must_use, deprecated)]
   |         ^^^^^^^^^^^^^^^
